src/cli-client/internals/cli/cli_upload.go

```diff
@@ -0,0 +1,150 @@
+	}
+	triggerUploadReleases(releases)
+	return nil
+}
+
+func checkMissingFields(release UploadRelease) []string {
+	var missing []string
+	if release.Track == "" {
+		missing = append(missing, "track")
+	}
+	if len(release.Risks) == 0 {
+		missing = append(missing, "risks")
+	}
+	if release.EndOfLife == "" {
+		missing = append(missing, "eol")
+	}
+	return missing
+}
+
+// parseUploadReleases parses the release arguments and returns a list of UploadRelease structs.
+// The release arguments are expected to be in the format "track=<track>,risks=<risk1>[,<risk2>...],eol=yyyy-mm-dd".
+// Multiple release arguments can be passed, separated by spaces.
+func parseUploadReleases(args []string) ([]UploadRelease, error) {
+	releases := make([]UploadRelease, 0)
+	regex := regexp.MustCompile(`(\w+)=(([^,=]*?,)*)`)
+
+	for _, origArgStr := range args {
+		// Append a "," to the end of the string to enable a simpler regex matching key-value pairs
+		argStr := origArgStr + ","
+		var release UploadRelease
+		matches := regex.FindAllStringSubmatch(argStr, -1)
+		if matches == nil {
+			return nil, fmt.Errorf("invalid argument: %s", origArgStr)
+		}
+
+		for _, part := range matches {
+			if part == nil || len(part) < 3 {
+				return nil, fmt.Errorf("invalid argument: %s", origArgStr)
+			}
+			key := part[1]
+			valueString := strings.TrimRight(part[2], ",")
+			values := strings.Split(valueString, ",")
+
+			switch key {
+			case "track":
+				if len(values) != 1 || values[0] == "" {
+					return nil, fmt.Errorf("invalid track value: %s", valueString)
+				}
+				if release.Track != "" {
+					return nil, fmt.Errorf("duplicated value for track")
+				}
+				release.Track = values[0]
+			case "risks":
+				for _, risk := range values {
+					if !slices.Contains(riskOptions, risk) {
+						return nil, fmt.Errorf("invalid risk value: %s", risk)
+					}
+				}
+				if len(release.Risks) > 0 {
+					return nil, fmt.Errorf("duplicated value for risks")
+				}
+				release.Risks = values
+			case "eol":
+				if len(values) != 1 || values[0] == "" {
+					return nil, fmt.Errorf("invalid eol value: %s", valueString)
+				}
+				eol, err := validateAndFormatDate(values[0])
+				if err != nil {
+					return nil, fmt.Errorf("EOL with wrong data formats: %v", err)
+				}
+				if release.EndOfLife != "" {
+					return nil, fmt.Errorf("duplicated value for eol")
+				}
+				release.EndOfLife = eol
+			default:
+				return nil, fmt.Errorf("invalid key-value pair: %s=%s", key, valueString)
+			}
+		}
+
+		// Check if all required fields are present
+		if missing := checkMissingFields(release); missing != nil {
+			return nil, fmt.Errorf("missing fields: %s", strings.Join(missing, ", "))
+		}
+
+		releases = append(releases, release)
+	}
+	if len(releases) == 0 {
+		return nil, fmt.Errorf("no release track specified, no build will be triggered")
+	}
+	return releases, nil
+}
+
+func triggerUploadReleases(releases []UploadRelease) {
```

Comment:
Need some error returns here. Lots of things within this function can go wrong it seems.

Reply:
Fixed in [db12a9a](https://github.com/canonical/oci-factory/pull/198/commits/db12a9a004a5879e6bf79da7e3f0a5d73025c759). All errors are handled internally as exiting or panicing.